packages/angular/src/library/jsonforms.component.ts

```diff
@@ -126,11 +127,32 @@ export class JsonFormsOutlet
       renderer !== undefined &&
       renderer.tester(uischema, schema, testerContext) !== -1
     ) {
-      bestComponent = renderer.renderer;
+      if (renderer.renderer instanceof Promise) {
+        renderer.renderer.then((resolvedRenderer) => {
+          bestComponent = resolvedRenderer;
+          const componentFactory =
+            this.componentFactoryResolver.resolveComponentFactory(
+              bestComponent
+            );
+          this.renderComponent(componentFactory, uischema, schema, props);
+        });
+        return;
+      } else {
+        bestComponent = renderer.renderer;
+      }
     }
 
     const componentFactory =
       this.componentFactoryResolver.resolveComponentFactory(bestComponent);
+    this.renderComponent(componentFactory, uischema, schema, props);
```

Comment:
For async components, this will lead to the `UnknownRenderer` being rendered first which is misleading and might look like an error to the user. Instead we should render a `LoadingRenderer` or something similar.